facade/src/main/java/org/jboss/pnc/facade/providers/ProductMilestoneProviderImpl.java

```diff
@@ -113,28 +113,19 @@ public ProductMilestoneProviderImpl(
     @Override
     public ProductMilestone update(String id, ProductMilestone restEntity) {
         validateBeforeUpdating(id, restEntity);
-
         org.jboss.pnc.model.ProductMilestone milestoneInDb = repository.queryById(Integer.valueOf(id));
-        org.jboss.pnc.model.ProductMilestone milestoneRestDb = mapper.toEntity(restEntity);
-
         // we can't modify milestone if it's already released
         if (milestoneInDb.getEndDate() != null) {
             log.info("Milestone is already closed: no more modifications allowed");
             throw new RepositoryViolationException("Milestone is already closed! No more modifications allowed");
         }
 
         log.debug("Updating milestone for id: {}", id);
-        milestoneRestDb.setId(Integer.valueOf(id));
-
-        // make sure that user cannot set the 'endDate' via the REST API
-        // this should only be set after the release process is successful
-        milestoneRestDb.setEndDate(milestoneInDb.getEndDate());
-        // Make sure that user cannot change the product release of the milestone. This is set on release creation.
-        milestoneRestDb.setProductRelease(milestoneInDb.getProductRelease());
+        ((ProductMilestoneMapper) mapper).mergeEntity(restEntity, milestoneInDb);
```

Comment:
Don't mind the cast here, it is for this prototype only. I would add the method to the root interface.